pkg/operator/rules.go

```diff
@@ -73,6 +73,30 @@ func NewPrometheusRuleSelector(ruleFormat RuleConfigurationFormat, version strin
 func (prs *PrometheusRuleSelector) generateRulesConfiguration(promRule *monitoringv1.PrometheusRule) (string, error) {
 	logger := log.With(prs.logger, "prometheusrule", promRule.Name, "prometheusrule-namespace", promRule.Namespace)
 	promRuleSpec := promRule.Spec
+
+	promRuleSpec = prs.sanitizePrometheusRulesSpec(promRuleSpec)
```

Comment:
we need to add/pass a logger parameter

Reply:
we have logger in prometheusRuleSelector, wont that be enough?